hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/ExpressionEvalContext.java

```diff
@@ -67,8 +74,29 @@ static ExpressionEvalContext from(Context ctx) {
             return new ExpressionEvalContextImpl(
                     arguments,
                     new DefaultSerializationServiceBuilder().build(),
-                    Util.getNodeEngine(ctx.hazelcastInstance()));
+                    Util.getNodeEngine(ctx.hazelcastInstance())
+            );
+        }
+    }
+
+    static ExpressionEvalContext createContext(
+            @Nonnull List<Object> arguments,
+            @Nonnull NodeEngine nodeEngine,
+            @Nonnull InternalSerializationService iss,
+            @Nullable SqlSecurityContext ssc) {
+        ExpressionEvalContext newEec = new ExpressionEvalContextImpl(arguments, iss, nodeEngine);
```

Comment:
add `SqlSecurityContext` to constructor so you can get rid of `setSecurityContext` method

Reply:
https://github.com/hazelcast/hazelcast/pull/25477/commits/47d30cf99a573360018f72c4b46883a121b223a1